src/builders/types.ts

```diff
@@ -11,7 +11,10 @@ export type ComponentCallback<
 	T extends ComponentInteraction | StringSelectMenuInteraction = ComponentInteraction | StringSelectMenuInteraction,
 > = (interaction: T, stop: ComponentStopCallback, refresh: ComponentRefreshCallback) => any;
 export type ComponentFilterCallback<T = ComponentInteraction> = (interaction: T) => any;
-export type ComponentStopCallback = (reason?: string, refresh?: ComponentRefreshCallback) => any;
+export type ComponentStopCallback = (
+	reason?: 'messageDelete' | 'channelDelete' | 'guildDelete' | (string & {}),
+	refresh?: ComponentRefreshCallback,

```

Comment:
This type should cover with an overload that if they are the first options, refresh does not exist